### PR TITLE
 Take into account ruleSeverity for TSLint errors

### PR DESCRIPTION
**Pull request by gexplorer**
All rules from TSLint where generating warnings in flycheck. I added the *ruleSeverity* parsing to *level* so TSLint errors generate flycheck errors instead of warnings

### CODE REVIEW

**Comment by gexplorer** [comment]
I already firmed the CLA, what should I do to get the pull request accepted?

File: flycheck.el [hunk 1]
```diff
@@ -5841,7 +5841,12 @@ about TSLint."
                  (flycheck-error-new-at
                   (+ 1 .startPosition.line)
                   (+ 1 .startPosition.character)
-                  'warning .failure
+                  (pcase .ruleSeverity
+                    (`"ERROR"   'error)
+                    (`"WARNING" 'warning)
+                    ;; Default to error
```
Comment (cpitclaudel) [inline flycheck.el]: No need for this comment; I think the code is clear without it :) Also, no need to quote strings in `pcase` patterns.

**Comment by cpitclaudel** [comment]
Thanks for your PR! This looks good. I've requested a few small changes, and then we can merge.

**Comment by cpitclaudel** [comment]
Also, shouldn't the default be warning, since that's what the behavior used to be?

**Comment by fmdkdd** [comment]
I've pushed a fix for the unrelated CI errors to master.  You will have to rebase on top for the CI to pass.

**Comment by fmdkdd** [comment]
Argh, seems like a recent change to `ess` broke Emacs 24.5 compatibility by using the new `project` built-in library instead of `projectile`.

I've disabled `ess` with an Emacs 24 specific Cask file for now.

**Comment by gexplorer** [comment]
Ok, thanks for the feedback, I'm an absolute beginner in elisp, I'll fix them as soon as possible.

**Comment by cpitclaudel** [comment]
Rebased and cleaned up in #1686

**Comment by cpitclaudel** [comment]
Rebased and cleaned up in #1686

**Pull request closed without merging**